spec/before.js

```diff
@@ -1,2 +1,5 @@
 // Trick Leaflet into believing we have a touchscreen (for desktop)
-if (window.TouchEvent) { window.ontouchstart = function(){} };
+if (window.TouchEvent) { window.ontouchstart = function () {}; }
+
+// allow tests for `detectRetina`
+window.devicePixelRatio = 2;
```

Comment:
You should not change this globally.

Reply:
Do you have any recommendations as to how to set retina to true at runtime for the tilelayer?

Reply:
I don't really see a good way to mock L.Browser.retina, since it's just an exported var/const, and I can't import anything in the spec file so that I can mock it with sinon.

Reply:
Looks like I might be able to install karma-requirejs in order to mock L.Browser: https://medium.com/@BoweiHan/setup-for-testing-amd-modules-with-karma-requirejs-and-mocha-2be3931c6a72

Is that something we would want to do?

Reply:
Just tried installing requirejs and karma-requirejs and it is a pain to configure.  Unless someone else knows better, I won't be able to add that for mocking L.Browser.

Reply:
> Do you have any recommendations as to how to set retina to true at runtime for the tilelayer?

That is good question. And I see now that our suite lacks `detectRetina` tests completely(!).
ATM I can propose such approach:
1) introduce `skipIfNoRetina`/`skipIfRetina` helpers, similar to that: https://github.com/Leaflet/Leaflet/blob/bc918d4bdc2ba189807bc207c77080fb41ecc196/spec/suites/SpecHelper.js#L57-L60
2) Setup separate browser config with proper retina params, similar to that: https://github.com/Leaflet/Leaflet/blob/bc918d4bdc2ba189807bc207c77080fb41ecc196/spec/karma.conf.js#L108-L114
(Actually I do not know if FF has such option, need to inspect it)

Reply:
> Looks like I might be able to install karma-requirejs in order to mock L.Browser:

I am not sure that it is possible to achieve in this way.
But you are right: that can be really useful, and not only for this test, but for many other.
I believe that we should refactor `L.Browser` to be mutable (and then just set arbitrary `L.Browser.retina` in runtime).

Reply:
I can definitely look into making L.Browser mutable.  At first glance that does not seem difficult.

Reply:
@IvanSanchez @mourner 
What do you think, is it good idea to make `L.Browser` mutable?